kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -218,4 +224,32 @@ public void close() throws IOException {
             }
         };
     }
+
+    /**
+     * Given a checkpoint file, retrieve all the files that are part of the same checkpoint.
+     * <p>
+     * This is done by looking at the file name and finding all the files that have the same
+     * version number.
+     */
+    private CloseableIterator<FileStatus> retrieveRemainingCheckpointFiles(
+            FileStatus checkpointFile,
+            long version) {
+
+        // Filter out all the files that are not part of the same checkpoint
+        final List<FileStatus> checkpointFiles = new ArrayList<>();
+
+        // Add the already retrieved checkpoint file to the list.
+        checkpointFiles.add(checkpointFile);
+
+        while (filesList.peek() != null &&
+                filesList.peek().getPath().endsWith(".parquet") &&
+                checkpointVersion(filesList.peek().getPath()) == version) {
+            checkpointFiles.add(filesList.pop());
+        }
+
+        // Remove the files from the list
+        filesList.removeAll(checkpointFiles);
```

Comment:
why do you need removeAll when you are already doing `filesList.pop()`?

Reply:
not needed.